### PR TITLE
dynamic subscriptions to layers and tables

### PR DESCRIPTION
**Pull request by KatKatKateryna**
Follors the PR https://github.com/specklesystems/speckle-sharp/pull/3423 and adds dynamic subscriptions to the data source changes of layers and tables

### CODE REVIEW

**Review by oguzhankoral — APPROVED** [review]
There was an issue on registration. Probably it was missed today on @adamhathcock's PR. I've fixed it.

Tested the data tables on change detection and works great. Codewise, send binding is bubbling but we have a ticket to simplify it on every connector https://spockle.atlassian.net/browse/CNX-9576

File: DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs [hunk 1]
```diff
@@ -57,6 +62,111 @@ private void SubscribeToArcGISEvents()
...
+
+  private void SubscribeToAnyDataSourceChange(Table layerTable)
+  {
+    RowCreatedEvent.Subscribe(
+      (args) =>
+      {
+        OnRowChanged(args);
+      },
+      layerTable
+    );
```
Comment (oguzhankoral) [inline DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: Hmm, what if we have saved layer tables on the document? Should we subscribe each table at the initialization?

Comment (KatKatKateryna) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: Yes, it's called on the ActiveMapView change event, which means whenever we open a map all the saved layers will be subscribed

Comment (oguzhankoral) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: Super!

Comment (KatKatKateryna) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: LayersAdded & TablesAdded events are catching up when we add layers to the document. Or did you mean something else?

Comment (KatKatKateryna) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: Ah yes, MapViewChanged subscribes to every layer existing in the Map which we have opened/ switched to. But definitely worth testing for edge cases

Comment (oguzhankoral) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs:127]: Then we are good for now, when you are back, we can investigate edge cases.

**Pull request merged**